### PR TITLE
Rename path to model_path and make it absolute, add version_path

### LINKED ISSUE #48: Rename path to model_path and make it absolute, add version_path

### PR DESCRIPTION
**Pull request by robzor92**
Fixes #48

### CODE REVIEW

File: python/hsml/model.py [hunk 1]
```diff
@@ -297,14 +297,16 @@ def experiment_project_name(self, experiment_project_name):
...
+    def model_path(self):
+        """path of the model with version folder omitted. Resolves to /Projects/{project_name}/Models/{name}"""
+        return "/Projects/{}/Models/{}".format(self.project_name, self.name)
+
+    @property
+    def version_path(self):
+        """path of the model including version folder. Resolves to /Projects/{project_name}/Models/{name}/{version}"""
+        return "/Projects/{}/Models/{}/{}".format(
+            self.project_name, self.name, str(self.version)
+        )
```
Comment (javierdlrm) [inline python/hsml/model.py:309]: Could we reuse _model_path_?

```suggestion
        return "{}/{}".format(self.model_path, str(self.version))
```

File: python/hsml/engine/model_engine.py [hunk 2]
```diff
@@ -352,18 +352,18 @@ def read_json(self, model_instance, artifact):
     def delete(self, model_instance):
         self._engine.delete(model_instance)
 
-    def add_tag(self, model, name, value):
+    def add_tag(self, model_instance, name, value):
```
Comment (javierdlrm) [inline python/hsml/engine/model_engine.py:355]: Just out of curiosity, why do you think `model_instance` is better?

**Review by javierdlrm — CHANGES_REQUESTED** [review]

File: python/hsml/engine/model_engine.py [hunk 3]
```diff
@@ -157,7 +157,7 @@ def _set_model_version(
         return model_instance
 
     def _build_artifact_path(self, model_instance, artifact):
-        artifact_path = model_instance.path + "/" + artifact
+        artifact_path = "{}/{}".format(model_instance.version_path, artifact)
```
Comment (javierdlrm) [inline python/hsml/engine/model_engine.py:160]: Is the artifact path built correctly? Just to make sure that we aren't missing the extra _/Artifacts/_ level.
Artifacts are stored in the `Artifacts` directory in the model version directory. For example, `/Projects/<p>/Models/<model_name>/<model_version>/Artifacts/<artifact_version>/<zip_file>.zip`.

Comment (robzor92) [reply python/hsml/engine/model_engine.py:160]: Okay so bad naming, by artifact I mean for example the `environment.yml` file or `input_example.json`, this is not related to those artifacts. I'll call it resource instead.

File: python/hsml/engine/model_engine.py (hunk 2 above)
Comment (robzor92) [reply python/hsml/engine/model_engine.py:355]: To make it consistent with the current naming when referring to a model in model_engine.py, for me it doesn't really matter if we call it model or model_instance, they are equal to me.

File: python/hsml/model.py (hunk 1 above)
Comment (robzor92) [reply python/hsml/model.py:309]: That's a good idea I'll do that

File: python/hsml/engine/model_engine.py (hunk 2 above)
Comment (javierdlrm) [reply python/hsml/engine/model_engine.py:355]: Ok! To me `model_instance `is the same as `model`, but with an extra word. But it's fine for now

**Review by javierdlrm — APPROVED** [review]

**Pull request merged**